Pong/main.c

```diff
@@ -14,15 +14,317 @@ const char* WINDOW_TITLE = "Pong";
+    SDL_Rect rect = {
+        .x = player->x,
+        .y = player->y - PLAYER_HEIGHT / 2,
+        .w = PLAYER_WIDTH,
+        .h = PLAYER_HEIGHT,
+    };
+    SDL_RenderFillRect(renderer, &rect);
+}
+//#endregion
+
+#pragma mark Court
+//#region Court
+const int SCORE_POINTS = 100;
+
+void court_incrementPlayerScore(SDL_Window *window, Player *player, int points) {
+    player->score += points;
+    
+    char *fmt = "Player 1: %d | Player 2: %d";
+    int len = snprintf(NULL, 0, fmt, player1.score, player2.score);
+    char buf[len + 1];
+    snprintf(buf, len + 1, fmt, player1.score, player2.score);
+    SDL_SetWindowTitle(window, buf);
+}
+
+// Bugs:
+// 1. Corners make the ball travel parallel to the length of the paddle.
+// 2. Sides sometimes make the ball clip into the paddle.
+// TODO: Fix this. https://stackoverflow.com/questions/56606799/how-to-detect-the-side-on-which-collision-occured
+void court_bounceBallFromPlayer(Ball *ball, Player *player) {
+    // Left side collision
+    if (
+        ball->x + BALL_SIZE / 2 <= player->x - PLAYER_WIDTH / 2
+    ) {
+        ball->xSpeed = -(ball->xSpeed);
+        return;
+    }
+    
+    // Right side collision
+    if (
+        ball->x - BALL_SIZE / 2 >= player->x + PLAYER_WIDTH / 2
+    ) {
+        ball->xSpeed = -(ball->xSpeed);
+        return;
+    }
+    
+    // Top side collision
+    if (
+        ball->y - BALL_SIZE / 2 <= player->y + PLAYER_HEIGHT / 2
+    ) {
+        ball->ySpeed = -(ball->ySpeed);
+        return;
+    }
+    
+    // Bottom side collision
+    if (
+        ball->x - BALL_SIZE / 2 >= player->x + PLAYER_WIDTH / 2
+    ) {
+        ball->ySpeed = -(ball->ySpeed);
+        return;
+    }
+}
+
+void court_updateCourt(SDL_Window *window, Ball *ball, float timeElapsed) {
+    const Uint8* keyboardState = SDL_GetKeyboardState(NULL);
+    
+    if (keyboardState[SDL_SCANCODE_SPACE]) {
+        served = true;
+    }
+    
+    int ballHalfSize = ball->size / 2;
+    
+    // Check if ball rect overlaps with either player's rect
+    SDL_Rect ballRect = {
+        .x = ball->x - ballHalfSize,
+        .y = ball->y - ballHalfSize,
+        .w = ball->size,
+        .h = ball->size,
+    };
+    
+    SDL_Rect player1Rect = {
+        .x = P1_X,
+        .y = (int)(player1.y) - PLAYER_HEIGHT / 2,
+        .w = PLAYER_WIDTH,
+        .h = PLAYER_HEIGHT,
+    };
+    
+    if (SDL_HasIntersection(&ballRect, &player1Rect)) {
+        // TODO: Branch out here. When the ball collides with the sides instead of the face the paddle, flip ySpeed instead of xSpeed.
+        // Ball goes to the right, but doesn't change vertical direction.
+        court_bounceBallFromPlayer(ball, &player1);
+    }
+    
+    SDL_Rect player2Rect = {
+        .x = P2_X,
+        .y = (int)(player2.y) - PLAYER_HEIGHT / 2,
+        .w = PLAYER_WIDTH,
+        .h = PLAYER_HEIGHT,
+    };
+    
+    if (SDL_HasIntersection(&ballRect, &player2Rect)) {
```

Comment:
https://github.com/KajiTetsushi/Pong/issues/13 switch to [`SDL_HasIntersectionF`](https://wiki.libsdl.org/SDL_HasIntersectionF).

Reply:
fixes #13